### ISSUE TITLE
[Camera Support]: go2rtc restream jsmpeg corrupted

### ISSUE THREAD

**Issue by amosyuen** [blakeblackshear/frigate#5958]
### Describe the problem you are having

Hi love all the great new features in v0.12. Was trying to setup go2rtc restream, it works fine with my Eufy C24 camera, but with my Wyze v2 cameras, when viewing the jsmpeg stream, it initially looks fine, but becomes corrupted usually within the next minute or two. 
![image](https://user-images.githubusercontent.com/422794/230794476-c215f258-7887-4d2b-bf17-53b0fbf305f1.png). Then after several more minutes becomes a white screen. 
![image](https://user-images.githubusercontent.com/422794/230794780-230676a6-45c8-414d-9441-fa127c5b1d8f.png)

Setting up the wyze camera without using restream, and using the RTSP directly works. I notice in the go2rtc restream setup, the baby camera had error `baby exceeded fps limit`.

In my setup baby camera is using the restream, garage camera is using direct RTSP. Both are wyze cameras.
Kitchen camera is a Eufy cam and works fine with restream.

### Version

0.12.0-BC16AD1

### Frigate config file

```yaml
detectors:
  cpu1:
    type: cpu
    num_threads: 1
  cpu2:
    type: cpu
    num_threads: 1
  cpu3:
    type: cpu
    num_threads: 1

detect:
  width: 1920
  height: 1080

motion:
  # Optional: The threshold passed to cv2.threshold to determine if a pixel is different enough to be counted as motion. (default: shown below)
  # Increasing this value will make motion detection less sensitive and decreasing it will make motion detection more sensitive.
  # The value should be between 1 and 255.
  threshold: 35
  # Optional: Minimum size in pixels in the resized motion image that counts as motion (default: 30)
  # Increasing this value will prevent smaller areas of motion from being detected. Decreasing will
  # make motion detection more sensitive to smaller moving objects.
  # As a rule of thumb:
  #  - 15 - high sensitivity
  #  - 30 - medium sensitivity
  #  - 50 - low sensitivity
  contour_area: 30
  # Optional: Alpha value passed to cv2.accumulateWeighted when averaging the motion delta across multiple frames (default: shown below)
  # Higher values mean the current frame impacts the delta a lot, and a single raindrop may register as motion.
  # Too low and a fast moving person wont be detected as motion.
  delta_alpha: 0.2
  # Optional: Alpha value passed to cv2.accumulateWeighted when averaging frames to determine the background (default: shown below)
  # Higher values mean the current frame impacts the average a lot, and a new object will be averaged into the background faster.
  # Low values will cause things like moving shadows to be detected as motion for longer.
  # https://www.geeksforgeeks.org/background-subtraction-in-an-image-using-concept-of-running-average/
  frame_alpha: 0.2
  # Optional: Height of the resized motion frame  (default: 50)
  # This operates as an efficient blur alternative. Higher values will result in more granular motion detection at the expense
  # of higher CPU usage. Lower values result in less CPU, but small changes may not register as motion.
  frame_height: 50
  # Optional: Delay when updating camera motion through MQTT from ON -> OFF (default: shown below).
  mqtt_off_delay: 1

objects:
  filters:
    person:
      # Optional: minimum width*height of the bounding box for the detected object (default: 0)
      #min_area: 5000
      # Optional: maximum width*height of the bounding box for the detected object (default: 24000000)
      #max_area: 100000
      # Optional: minimum width/height of the bounding box for the detected object (default: 0)
      #min_ratio: 0.5
      # Optional: maximum width/height of the bounding box for the detected object (default: 24000000)
      #max_ratio: 2.0
      # Optional: minimum score for the object to initiate tracking (default: shown below)
      min_score: 0.4
      # Optional: minimum decimal percentage for tracked object's computed score to be considered a true positive (default: shown below)
      threshold: 0.75

snapshots:
  enabled: True
  timestamp: False
  bounding_box: True
  retain:
    default: 7
    objects:
      person: 7
  quality: 100
  height: 500

go2rtc:
  streams:
    # Restream camera
    baby:
      - rtsp://<redacted>:<redacted>@192.168.0.134/live
      - "ffmpeg:baby#audio=opus"

    # Non-restream camera
    garage:
      - rtsp://<redacted>:<redacted>@192.168.0.125/live
      - "ffmpeg:garage#audio=opus"

    kitchen:
      - rtsp://<redacted>:<redacted>@192.168.0.114/live0
      - "ffmpeg:kitchen#audio=opus"
      
  webrtc:
    candidates:
      - 192.168.0.254:8555
      - stun:8555

cameras:
  # Restream
  baby:
    detect:
      stationary:
        interval: 25
        threshold: 500
    ffmpeg:
      inputs:
        - path: rtsp://localhost:8554/baby
          input_args: preset-rtsp-restream
          roles:
            - detect
    motion:
      mask:
        - &wyze_time_mask 1374,1080,1874,1080,1875,1004,1374,1000
    mqtt: &mqtt_config
      timestamp: False
      bounding_box: False
      crop: True
      quality: 100
      height: 500

  # Non-restream
  garage:
    ffmpeg:
      inputs:
        - path: rtsp://<redacted>:<redacted>@192.168.0.125/live
          roles:
            - detect
    motion:
      mask:
        - *wyze_time_mask
    mqtt: *mqtt_config

  kitchen:
    ffmpeg:
      inputs:
        - path: rtsp://localhost:8554/kitchen
          input_args: preset-rtsp-restream
          roles:
            - detect
    #detect:
    #  width: 2304
    #  height: 1296
    motion:
      mask:
        - &eufy_indoor_time_mask 1883,67,1886,0,1374,0,1370,65
        - &tv_mask 2248,309,2304,99,2082,0,2033,180
    mqtt: *mqtt_config
    objects:
      filters:
        person:
          mask:
            - *tv_mask
    zones:
      kitchen_zone:
        coordinates: 0,1296,2304,1296,2304,679,1682,282,1270,356,1301,0,0,0
      living_room_zone:
        coordinates: 1831,0,2304,0,2304,577,1629,149,1798,133
```

### Relevant log output

```shell
Frigate
===

2023-04-09 20:12:24.060942846  [INFO] Starting Frigate...
2023-04-09 20:12:31.558803731  [2023-04-09 13:12:31] frigate.app                    INFO    : Starting Frigate (0.12.0-bc16ad1)
2023-04-09 20:12:31.643254617  [2023-04-09 13:12:31] frigate.app                    INFO    : Creating directory: /tmp/cache
2023-04-09 20:12:31.649996540  [2023-04-09 13:12:31] peewee_migrate                 INFO    : Starting migrations
2023-04-09 20:12:31.668316950  [2023-04-09 13:12:31] peewee_migrate                 INFO    : There is nothing to migrate
2023-04-09 20:12:31.776859843  [2023-04-09 13:12:31] detector.cpu1                  INFO    : Starting detection process: 300
2023-04-09 20:12:31.806056091  [2023-04-09 13:12:31] frigate.app                    INFO    : Output process started: 305
2023-04-09 20:12:31.835781821  [2023-04-09 13:12:31] frigate.detectors              WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
2023-04-09 20:12:31.835790818  [2023-04-09 13:12:31] detector.cpu2                  INFO    : Starting detection process: 301
2023-04-09 20:12:31.835796499  [2023-04-09 13:12:31] frigate.detectors              WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
2023-04-09 20:12:31.852415063  [2023-04-09 13:12:31] detector.cpu3                  INFO    : Starting detection process: 303
2023-04-09 20:12:31.867577313  [2023-04-09 13:12:31] frigate.detectors              WARNING : CPU detectors are not recommended and should only be used for testing or for trial purposes.
2023-04-09 20:12:31.884463645  [2023-04-09 13:12:31] frigate.app                    INFO    : Camera processor started for baby: 310
2023-04-09 20:12:31.890304571  [2023-04-09 13:12:31] frigate.app                    INFO    : Camera processor started for garage: 311
2023-04-09 20:12:31.931082382  [2023-04-09 13:12:31] frigate.app                    INFO    : Camera processor started for kitchen: 312
2023-04-09 20:12:31.955949306  [2023-04-09 13:12:31] frigate.app                    INFO    : Capture process started for baby: 314
2023-04-09 20:12:31.990564270  [2023-04-09 13:12:31] frigate.app                    INFO    : Capture process started for garage: 316
2023-04-09 20:12:32.014942460  [2023-04-09 13:12:32] frigate.app                    INFO    : Capture process started for kitchen: 319
2023-04-09 20:13:22.091870129  [2023-04-09 13:13:22] watchdog.baby                  INFO    : baby exceeded fps limit. Exiting ffmpeg...
2023-04-09 20:13:22.092960916  [2023-04-09 13:13:22] watchdog.baby                  INFO    : Waiting for ffmpeg to exit gracefully...
2023-04-09 20:13:22.502634582  [2023-04-09 13:13:22] frigate.video                  ERROR   : garage: Unable to read frames from ffmpeg process.
2023-04-09 20:13:22.505189348  [2023-04-09 13:13:22] frigate.video                  ERROR   : garage: ffmpeg process is not running. exiting capture thread...
2023-04-09 20:13:32.120122115  [2023-04-09 13:13:32] watchdog.garage                ERROR   : Ffmpeg process crashed unexpectedly for garage.
2023-04-09 20:13:32.120129390  [2023-04-09 13:13:32] watchdog.garage                ERROR   : The following ffmpeg logs include the last 100 lines prior to exit.
2023-04-09 20:13:32.120133237  [2023-04-09 13:13:32] ffmpeg.garage.detect           ERROR   : Guessed Channel Layout for Input Stream #0.1 : mono
2023-04-09 20:13:32.120136889  [2023-04-09 13:13:32] ffmpeg.garage.detect           ERROR   : [h264 @ 0x5620d20fe8c0] error while decoding MB 28 25, bytestream -7
2023-04-09 20:13:32.120290671  [2023-04-09 13:13:32] ffmpeg.garage.detect           ERROR   : rtsp://*:*@192.168.0.125/live: Connection timed out
2023-04-09 20:13:32.120295505  [2023-04-09 13:13:32] ffmpeg.garage.detect           ERROR   : rtsp://*:*@192.168.0.125/live: corrupt decoded frame in stream 0
2023-04-09 20:13:32.120770301  [2023-04-09 13:13:32] ffmpeg.garage.detect           ERROR   : rtsp://*:*@192.168.0.125/live: Connection timed out
2023-04-09 20:14:02.224946866  [2023-04-09 13:14:02] frigate.video                  ERROR   : baby: Unable to read frames from ffmpeg process.
2023-04-09 20:14:02.224954269  [2023-04-09 13:14:02] frigate.video                  ERROR   : baby: ffmpeg process is not running. exiting capture thread...
2023-04-09 20:14:06.874679279  [2023-04-09 13:14:06] watchdog.baby                  INFO    : FFmpeg did not exit. Force killing...


go2rtc
===
2023-04-09 20:12:24.060999247  [INFO] Preparing go2rtc config...
2023-04-09 20:12:24.227381422  [INFO] Got IP address from supervisor: 192.168.0.249
2023-04-09 20:12:24.308242670  [INFO] Got WebRTC port from supervisor: 8555
2023-04-09 20:12:25.831047642  [INFO] Not injecting WebRTC candidates into go2rtc config as it has been set manually
2023-04-09 20:12:25.914620928  [INFO] Starting go2rtc...
2023-04-09 20:12:26.098988110  13:12:26.098 INF go2rtc version 1.2.0 linux/amd64
2023-04-09 20:12:26.100653857  13:12:26.100 INF [api] listen addr=:1984
2023-04-09 20:12:26.102073551  13:12:26.101 INF [rtsp] listen addr=:8554
2023-04-09 20:12:26.103475620  13:12:26.103 INF [srtp] listen addr=:8443
2023-04-09 20:12:26.104957215  13:12:26.104 INF [webrtc] listen addr=:8555
2023-04-09 20:12:33.976546876  [INFO] Starting go2rtc healthcheck service...
2023-04-09 20:13:12.441109560  13:13:12.440 WRN github.com/AlexxIT/go2rtc/cmd/streams/producer.go:133 > error="read tcp 172.30.33.12:33552->192.168.0.134:554: i/o timeout" url=rtsp://<redacted>:<redacted>@192.168.0.134/live
```

### FFprobe output from your camera

```shell
# Babycamera gortc restream

[{"return_code":0,"stderr":"","stdout":{"programs":[],"streams":[{"avg_frame_rate":"15/1","codec_long_name":"H.264/AVC/MPEG-4AVC/MPEG-4part10","height":1080,"width":1920}]}}]


# Garage camera (direct RTSP)

[{"return_code":0,"stderr":"","stdout":{"programs":[],"streams":[{"avg_frame_rate":"15/1","codec_long_name":"H.264/AVC/MPEG-4AVC/MPEG-4part10","height":1080,"width":1920},{"avg_frame_rate":"0/0","bit_rate":"64000","codec_long_name":"PCMA-law/G.711A-law"}]}}]
```

### Operating system

HassOS

### Install method

HassOS Addon

### Coral version

CPU (no coral)

### Network connection

Wired

### Camera make and model

Wyze v2

**Comment by NickM-27** [comment]
there have been a number of issues with wyze cameras given their low quality stream, so that is not surprising. What happens if you use ffmpeg for the entire stream?

```yaml
go2rtc:
  streams:
    # Restream camera
    baby:
      - "ffmpeg:rtsp://<redacted>:<redacted>@192.168.0.134/live#video=copy#audio=copy#audio=opus"

    # Non-restream camera
    garage:
      - "ffmpeg:rtsp://<redacted>:<redacted>@192.168.0.125/live#video=copy#audio=copy#audio=opus"

    kitchen:
      - "ffmpeg:rtsp://<redacted>:<redacted>@192.168.0.114/live0#video=copy#audio=copy#audio=opus"
```

**Comment by amosyuen** [comment]
Tried it, it improved the stream, seems corruption is less drastic, but still periodically gets corrupted, fails, then restarts.

**Comment by NickM-27** [comment]
in that case I would suggest having the frigate stream point to the cameras directly

**Comment by amosyuen** [comment]
Yeah that's fine to do as a workaround, does this indicate there is something that go2rtc doesn't support with the wyze cam?

**Comment by NickM-27** [comment]
> Yeah that's fine to do as a workaround, does this indicate there is something that go2rtc doesn't support with the wyze cam?

The wyze cam does not have a good rtsp stream, it does not follow the rtsp protocol and has many errors in the stream. Go2rtc does not handle these errors as well as some other software

**Comment by amosyuen** [comment]
Thanks for explaining, maybe I'll look into getting better cameras.

**Comment by my-umd** [comment]
I have a couple of Wyze v2 on rtsp firmware (version 4.28.4.41) using restreaming.  I watched about 5 minutes of jsmpeg live view from one of my v2s, seems fairly stable.  My observation was that v2 is not good when using MSE or webrtc.

Comparing your config with mine, I see some differences:
1.  I have 'aac' audio, not 'opus'.
2.  I use loopback ip 127.0.0.1, not 'localhost'.
3.  I have rtsp username and password also in the restream.  I don't see them in your case.

I am not an expert.  Thus I am not sure if any of these make any difference.  Also, I am not viewing from Home Assistant.  I have a standalone docker.  I viewed jsmpeg live feed from Frigate Web UI.  I am on Frigate 0.12.0-DA3E197 (a.k.a., official 0.12.0 release).
